ginivision/src/main/java/net/gini/android/vision/analysis/AnalysisFragmentImpl.java

```diff
@@ -118,11 +121,13 @@ public void onProceedToNoExtractionsScreen(@NonNull final Document document) {
     private boolean mStopped;
     private boolean mAnalysisCompleted;
 
-
     AnalysisFragmentImpl(final FragmentImplCallback fragment, @NonNull final Document document,
             final String documentAnalysisErrorMessage) {
         mFragment = fragment;
         mMultiPageDocument = asMultiPageDocument(document);
+        // Tag the documents to be able to clean up the automatically parcelled data
+        ((GiniVisionDocument) document).setParcelableMemoryCacheTag(PARCELABLE_MEMORY_CACHE_TAG);
```

Comment:
Since `setParcelableMemoryCacheTag` is not part of `Document` the instance should be checked.